tests/tests/custom_authorities_operations_tests.cpp

```diff
@@ -0,0 +1,428 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <boost/test/unit_test.hpp>
+#include <graphene/chain/protocol/custom_authority.hpp>
+#include <graphene/chain/custom_authority_object.hpp>
+#include <graphene/chain/operation_type_to_id.hpp>
+
+#include "../common/database_fixture.hpp"
+
+using namespace graphene::chain;
+using namespace graphene::chain::test;
+using namespace graphene::app;
+
+struct custom_authorities_operations_fixture: database_fixture
+{
+   void create_custom_authority(const account_id_type& account, bool enabled, int operation_type)
+   {
+      custom_authority_create_operation op; 
+      op.account = account;
+      op.enabled = enabled;
+      op.valid_from = db.head_block_time() - 1;
+      op.valid_to = db.head_block_time() + 20;
+      op.operation_type = operation_type;
+      
+      trx.operations.push_back(op);
+      trx.validate();
+      
+      db.push_transaction(trx, ~0);
+      trx.operations.clear();
+   }
+   
+   void update_account( const account_object& account)
+   {
+      trx.operations.clear();
+      
+      account_update_operation op;
+      op.account = account.id;
+      op.active = account.active;
+      
+      trx.operations = {op};
+      trx.validate();
+      
+      db.push_transaction(trx, ~0);
+      trx.clear();
+   }
+};
+
+BOOST_FIXTURE_TEST_SUITE( custom_authorities_operations, custom_authorities_operations_fixture )
+
+BOOST_AUTO_TEST_CASE(get_custom_authorities_by_account_without_authorities) {
+   try {
+      auto dan = create_account("dan");
+      
+      BOOST_CHECK(db.get_custom_authorities_by_account(dan.id).empty());
+   } catch (fc::exception &e) {
+      edump((e.to_detail_string()));
+      throw;
+   }
+}
+
+BOOST_AUTO_TEST_CASE(get_custom_authorities_by_account_without_authorities_but_with_authorities_for_another_account) {
+   try {
+      auto dan = create_account("dan");
+      auto sam = create_account("sam");
+      
+      create_custom_authority(sam.id, true, operation_type_id_from_operation_type<transfer_operation>::value);
+      
+      auto authorities = db.get_custom_authorities_by_account(dan.id);
+      BOOST_REQUIRE_EQUAL(0, authorities.size());
+   } catch (fc::exception &e) {
+      edump((e.to_detail_string()));
+      throw;
+   }
+}
+
+BOOST_AUTO_TEST_CASE(create_custom_authority_operation_adds_authority_to_db) {
```

Comment:
Please add another test case that also tries to execute the transfer (or include in this one)

Reply:
Done?

Reply:
I've added tests for transfer to the bottom of this file. 
transaction_fails*, transaction_passes* tests.